app/src/main/java/com/javahelp/frontend/activity/ProviderRegistrationViewmodel.java

```diff
@@ -1,17 +1,199 @@
+    private MutableLiveData<Optional<RegisterResult>> registerResult = new MutableLiveData<>(Optional.empty());
+    private ProviderUserInfo providerUserInfo = new ProviderUserInfo("","","","");
+    private RegisterInteractor registerInteractor;
+
 
 
     public ProviderRegistrationViewmodel(@NonNull Application application) {
         super(application);
+        ISaltDataAccess salt = LambdaSaltDataAccessProvider.getInstance();
+        IRegisterDataAccess register = LambdaRegisterDataAccess.getInstance();
+        IPasswordHasher hasher = SHAPasswordHasher.getInstance();
+        registerInteractor = new RegisterInteractor(this, salt, register, hasher);
+    }
+
+    /**
+     * Sets the username for this {@link ProviderRegistrationViewmodel}
+     * @param username {@link String} username
+     */
+    public void setUsername(String username) {
+        this.username.setValue(username);
+    }
+
+    /**
+     * Sets the password for this {@link ProviderRegistrationViewmodel}
+     * @param password1 {@link String} password1 to use
+     */
+    public void setPassword1(String password1) {
+        this.password1.setValue(password1);
+    }
+
+    /**
+     * Sets the confirmed_password for this {@link ProviderRegistrationViewmodel}
+     * @param password2 {@link String} password2 to use
+     */
+    public void setPassword2(String password2) { this.password2.setValue(password2);}
+
+    /**
+     * Sets the email for this {@link ProviderRegistrationViewmodel}
+     * @param email {@link String} email to use
+     */
+    public void setEmail(String email) { this.email.setValue(email);}
+
+    /**
+     * Sets the firstname for this {@link ProviderRegistrationViewmodel}
+     * @param firstname {@link String} firstname to use
+     */
+    public void setFirstname(String firstname) { this.firstname.setValue(firstname);}
+
+    /**
+     * Sets the lastname for this {@link ProviderRegistrationViewmodel}
+     * @param lastname {@link String} lastname to use
+     */
+    public void setLastname(String lastname) { this.lastname.setValue(lastname);}
+
+    /**
+     * Sets the address for this {@link ProviderRegistrationViewmodel}
+     * @param address {@link String} address to use
+     */
+    public void setAddress(String address) { this.address.setValue(address);}
+
+    /**
+     * Sets the phone for this {@link ProviderRegistrationViewmodel}
+     * @param phone {@link String} phone to use
+     */
+    public void setPhone(String phone) { this.phone.setValue(phone);}
+
+    /**
+     * Sets whether the current provider is certified or not
+     * @param c whether certified
+     */
+    public void setCertified(boolean c) {certified.setValue(c);}
+
+    /**
+     * Sets the register error
+     * @param s {@link String} register error to set
+     */
+    public void setRegisterError(String s) {
+        registerResult.setValue(Optional.of(new RegisterResult(s)));}
+
+    /**
```

Comment:
In the `[Backend] (Provider) Register User Use Case` will bundle these together, you don't need to worry about doing it here (and also for `Clean Architecture` I think)

Reply:
I'm not sure I follow how the backend makes this redundant. Remember, we don't want any dependence between our frontend and our backend.